plugins/services/src/js/structs/Framework.js

```diff
@@ -66,6 +123,55 @@ module.exports = class Framework extends Application {
     return this._spec;
   }
 
+  getCheckStatus() {
+    const checkStatus = this.get("tasks").reduce((acc, cur) => {
+      if (cur.checkResult != null) {
+        const curStatusCode = findNestedPropertyInObject(
+          cur,
+          "checkResult.http.statusCode"
+        );
+        const accStatusCode =
+          findNestedPropertyInObject(acc, "checkResult.http.statusCode") ||
+          null;
+        if (curStatusCode > accStatusCode) {
+          return cur;
+        }
+      }
+
+      return acc;
+    }, false);
+
+    if (checkStatus == null) {
+      return false;
+    }
+
+    return checkStatus;
+  }
+
+  getStatus() {
```

Comment:
I'm finding the naming of this method as well as `getServiceStatus` a bit ambiguous. Hard to tell what the difference is without doing some digging (one returns string display name and the other an object)?

Reply:
I can't rename these functions since they are overwriting functions form application. But what I can do is add more documentation as in JS doc blocks.